library/src/main/java/com/chuckerteam/chucker/api/RetentionManager.kt

```diff
@@ -0,0 +1,101 @@
+package com.chuckerteam.chucker.api
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import com.chuckerteam.chucker.api.Chucker.LOG_TAG
+
+import com.chuckerteam.chucker.api.internal.data.repository.RepositoryProvider
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Class responsible of holding the logic for the retention of your HTTP transactions
+ * and your throwable. You can customize how long data should be stored here.
+ * @param context An Android Context
+ * @param retentionPeriod A [Period] to specify the retention of data. Default 1 week.
+ */
+class RetentionManager @JvmOverloads constructor(
+        context: Context,
+        retentionPeriod: Period = Period.ONE_WEEK
+) {
+
+    // The actual retention period in milliseconds (default to
+    private val period: Long
+    // How often the cleanup should happen
+    private val cleanupFrequency: Long
+    private val prefs: SharedPreferences
+
+    init {
+        period = toMillis(retentionPeriod)
```

Comment:
Those lines can be done in the property definition.

`private val period: Long = context.get......`